pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/TokenService.scala

```diff
@@ -0,0 +1,52 @@
+package uk.ac.wellcome.platform.transformer.mets.service
+
+import java.net.URI
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.stream.Materializer
+import com.github.dakatsuka.akka.http.oauth2.client.{Client, Config, GrantType}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class TokenService(url: String,
+                   clientId: String,
+                   secret: String,
+                   scope: String,
+                   initialDelay: FiniteDuration,
+                   interval: FiniteDuration)(implicit actorSystem: ActorSystem,
+                                             mat: Materializer,
+                                             ec: ExecutionContext) {
+  private val config = Config(
+    clientId = clientId,
+    clientSecret = secret,
+    site = URI.create(url),
+    tokenUrl = "/oauth2/token"
+  )
+
+  private val client = Client(config)
+  private val token = new AtomicReference[OAuth2BearerToken]
+
+  actorSystem.scheduler.schedule(initialDelay, interval)(refreshToken())
```

Comment:
It appears that this method is deprecated with most recent version of akka:

https://github.com/akka/akka/blob/d4ed47f3091c48f708bfc4a4327c4933e4390e2b/akka-actor/src/main/scala/akka/actor/Scheduler.scala#L393-L400

Reply:
oh thanks, I'll change it

Reply:
Actually, annoyingly the version of akka in our classpath doesn't have the suggested methods. We should fix it once we update akka but I'm reluctant to do it in this PR